### ISSUE TITLE
Unexpected panic while rearranging macro arguments

### ISSUE THREAD

**Issue by Xaeroxe** [alexpovel/srgn#71]
Invocation:

```
$ srgn --version
srgn 0.11.0

$ srgn --files '**/*.rs' --rust-query '((macro_invocation macro: (identifier) @name) @macro (#any-of? @name "error" "warn" "info" "debug" "trace" ))' '(?P<msg>.+);(?P<structure>.+)' '$structure,$msg'
```
Panic:

```
thread '<unnamed>' panicked at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/srgn-0.11.0/src/scoping/scope.rs:60:43:
begin <= end (23 <= 0) when slicing `error!( "server error"; "uuid" => %uuid, "error.user_msg" => %user_msg, src)`
```

Please let me know if more information is needed.

**Comment by alexpovel** [comment]
Hi! So the panic was simply a bug, which is fixed now. The syntax of `'(?P<msg>.+);(?P<structure>.+)' '$structure,$msg'` still doesn't work as intended yet, but as mentioned I'll be working on implementing that. In the meantime, to solve your immediate problem, perhaps [ast-grep](https://ast-grep.github.io/) can help you?

**Comment by alexpovel** [comment]
Hi @Xaeroxe !

Support for variables landed in `main` recently, at 7f6cfcbcef8f8d010de5b12df4d3e749b655d128 . That will make it into a release soon-ish. Then, an invocation like `'(?P<msg>.+);(?P<structure>.+)' '$structure,$msg'` should start working (see [here](https://github.com/alexpovel/srgn/tree/7f6cfcbcef8f8d010de5b12df4d3e749b655d128?tab=readme-ov-file#variables) for the docs).
